### PR TITLE
Friendly Environment policy

### PR DESCRIPTION
**Pull request by mflatt**
Continued from #105

### CODE REVIEW

**Comment by mflatt** [comment]
The intent of this revision is to simplify. Instead of setting up a group of moderators, this variant makes explicit that preserving a friendly environment is Racket management's responsibility as well as its goal.

**Comment by samth** [comment]
@lexi-lambda you had some useful comments on the original draft of this, and I'm interested in your thoughts on the revision.

File: www/index.html.pm [hunk 1]
```diff
@@ -231,6 +231,8 @@ Community
 
 ◊link["sfc.html"]{Software Freedom Conservancy}
 
+◊link["friendly.html"]{Friendly Enviornment policy}
```
Comment (bennn) [inline www/index.html.pm]: typo "Enviornment"

File: www/friendly.html.pm [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}
+}
+
+
+◊section[#:class "one-column-body-text"]{
+Friendly Environment Policy
+
+The Racket community aims to improve the world through programming. It started with the goal of introducing everyone to the wonderful world of program design, with a spirit of full inclusion and no emphasis on any specific group.  Over time it has grown into a full-fledged professional community with a well-known reputation for helpfulness and openness on its on-line communication channels. Racket project management wants to encourage and preserve that open exchange of ideas.
+
+This spirit requires an environment that enables all to participate without fear of personal harassment. We define harassment as unwelcome or hostile behavior, that is, behavior that focuses on people instead of ideas. ◊link["http://www.acm.org/sigs/volunteer_resources/officers_manual/anti-harassment-policy"]{The ACM's anti-harassment policy} lists a variety of specific unacceptable factors and behaviors. Racket project management considers responses such as “just joking,” or “teasing,” or being “playful” as unacceptable.
```
Comment (lexi-lambda) [inline www/friendly.html.pm]: This link does not go anywhere useful for me (I just get redirected to https://www.acm.org/special-interest-groups/volunteer-resources, which does not have any obvious link to the policy that I could find), which seems like a pretty big problem.

Comment (samth) [reply www/friendly.html.pm]: I think it's now at https://www.acm.org/about-acm/policy-against-harassment

Comment (mflatt) [reply www/friendly.html.pm]: There's an "ACM Policies" box on the lower right. But that took me a while to find, too, so probably a more direct link is better.

Comment (lexi-lambda) [reply www/friendly.html.pm]: Thanks! Since the link seems questionably stable, I think it would be better if the salient points were quoted or paraphrased on this page directly, but maybe it isn’t a big problem.

Comment (lexi-lambda) [reply www/friendly.html.pm]: > We define harassment as unwelcome or hostile behavior, that is, behavior that focuses on people instead of ideas.

I would reword this sentence. While I think it means well, as-worded it essentially says “do not criticize someone for their personal beliefs.” Though quite reasonable in a vacuum, this and similar sentiments (such as “be nice” or “be civil”) have sadly been used to take action against people who call attention to discrimination, especially if that discrimination took place in a different forum or community.

I don’t think anyone has any reason to worry about that problem in the Racket community, but those who benefit most from a Code of Conduct don’t have the luxury of already knowing that context. Most people will probably interpret it charitably, but it helps to leave no room for doubt.

I like the way [Rust’s Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct) phrases this:

> We will exclude you from interaction if you insult, demean or harass anyone. […] In particular, we don’t tolerate behavior that excludes people in socially marginalized groups. Private harassment is also unacceptable.

It’s short and to the point, but it’s also very clear.

Comment (mfelleisen) [reply www/friendly.html.pm]: I don't like the Rust sentence at all. It is overly aggressive and focused on specific groups. 

We care that every *individual* is treated properly, period. Much stronger, while not being aggressive againt, or focused on, any group.

Comment (jackfirth) [reply www/friendly.html.pm]: There is no way to properly address harassment without addressing the fact that some groups of people are far more likely to be harassed than others.

Comment (mfelleisen) [reply www/friendly.html.pm]: There is. Respect every individual. Every single one of them. The End.

Comment (lexi-lambda) [reply www/friendly.html.pm]: Matthias: I am not suggesting copying the Rust Code of Conduct wholesale. I do suggest incorporating the three sentences I quoted. Rewording them slightly is fine, but muddying their meaning is not.

The first sentence reads, “We will exclude you from interaction if you insult, demean or harass **anyone**.” It is very clear. You may not find the second sentence necessary, but it is hardly offensive, and it will give some people who particularly need it a feeling of additional safety.

With all due respect: don’t die on this particular hill. It doesn’t really affect you. I guarantee it matters quite a lot more to some people than you find it philosophically unnecessary.

Comment (mfelleisen) [reply www/friendly.html.pm]: It affects me in very serious ways, and that's precisely why I don't want to speak of goops, ever. As a non-liberal I am routinely attacked---including at a RacketCon a couple of years back. I don't seek help, I don't advertise this, I cope but I ascribe all of this to the group think that is so pervasive in the modern and especially the tech world. 

You have already exclude me from 'users'. If you wish to exclude me from other venues too, this is the way to go.

Comment (lexi-lambda) [reply www/friendly.html.pm]: > As a non-liberal I am routinely attacked---including at a RacketCon a couple of years back. I don't seek help, I don't advertise this, I cope but I ascribe all of this to the group think that is so pervasive in the modern and especially the tech world.

If you feel you have been harassed, it seems like this very document provides some recommendations on how that should be handled, with or without my suggestions.

But that is really immaterial. Clearly, you agree that “socially marginalized groups” should be able to participate in Racket community spaces without fear of harassment, seeing as they are a subset of “everyone.” Offering marginalized people—that is, people who are discriminated against more often *by the definition of the word “marginalized”*—extra reassurance that they are, in fact, welcome should not be a political issue. If you feel that sentence is an “attack” on “non-liberals,” I am not sure what to say.

Comment (jeapostrophe) [reply www/friendly.html.pm]: I heartily agree with Matthias. We want to foster respect and kindness and love and charity for all people regardless of their membership in any particular groups. No groups deserve more or less respect or protection. By including this sentence, we are favoring some over others.

Comment (AlexKnauth) [reply www/friendly.html.pm]: I agree with @lexi-lambda. I agree we should focus on respect for individual people... that's what the first sentence is for. It applies to people whether they're part of an often-marginalized group or not. But that doesn't make the next part unnecessary. I think we should give extra assurance that we don't allow harassment against people in marginalized groups in a follow-up sentence.

**Comment by mflatt** [comment]
I've pushed a revised proposal that removes the attempt to define "harassment" in one sentence and that instead appeals to an existing definition — like the Rust CoC does, but sticking here with the ACM policy. Along those lines, I removed the elaboration about "just joking".

We can't outsource our responsibility to understand and encourage acceptable behavior, but we don't need to innovate in this area. That's why I'm moving the PR further in the direction of relying on the ACM policy.

**Comment by spdegabrielle** [comment]
Thank you @mflatt @samth @jeapostrophe @mfelleisen @lexi-lambda and @AlexKnauth for engaging with this as I believe it is am important issue.

If you are being attacked then it is inevitable that other people **without** the requisite experience and coping skills will also be attacked.

I believe that if a member of the racket management team experiences harassment then they should take action, either by directly excluding the harasser, of if they are worried about conflict of interest, referrer it to another member of the management team to take action. Please tell me if you think I'm wrong as I believe that it the intention of the following line of the FOP; 

>Anyone witnessing or subject to unacceptable behavior should notify ◊link["team.html#management"]{Racket project management}

@lexi-lambda was **invited** to provide feedback, and she kindly took time out oh her day to to provide responses that included concerns, suggested text and explained the rationale. 

Please suggest a text that is clear and you are willing to enforce, because as with the actual instance of harassment above, a policy is pointless if those responsible are not willing to enforce it.

My apologies if I come across as curt or grumpy. This is just what I sound like when I am trying (and probably failing) to write clearly.

Once again, thank you Jay, Matthias and Alexis for taking the time to work on this.

Every day I see examples of the tireless work all three of you do in the Racket community. I believe this is because of your common desire to 'foster respect and kindness and love and charity for all people'.  All three of you have helped me many times.

Thanks again,

Stephen

PS Don't give up!

**Comment by rmculpepper** [comment]
The "just joking" text seems to be reworded but not removed. Was that the intended change?

**Comment by mflatt** [comment]
No, sorry — I pushed the wrong version. It should be as intended, now.

**Comment by mflatt** [comment]
Any objections to the latest version? If not, I plan to merge in a day or so.

File: www/index.html.pm (hunk 1 above)
Comment (jeapostrophe) [reply www/index.html.pm]: The typo is still there

**Comment by jeapostrophe** [comment]
I'm good, minus that typo.

**Comment by mflatt** [comment]
Thanks for the reminder – now fixed.

**Comment by mfelleisen** [comment]
Me too.

**Comment by spdegabrielle** [comment]
Just saw the FEP on the website! thank you @mflatt @mfelleisen @jeapostrophe
